build/Dockerfile.c7.layered

```diff
@@ -1,81 +1,158 @@
+        lz4-static \
+        mono-devel \
+        rh-git218 \
+        rh-python36 \
+        rh-python36-devel \
+        rh-ruby26 \
+        rpm-build \
+        unzip \
+        wget && \
+    yum clean all && \
+    rm -rf /var/cache/yum
 
 # install Ninja
-RUN cd /tmp && curl -L https://github.com/ninja-build/ninja/archive/v1.9.0.zip -o ninja.zip &&\
-    unzip ninja.zip && cd ninja-1.9.0 && scl enable devtoolset-8 -- ./configure.py --bootstrap && cp ninja /usr/bin &&\
-    cd .. && rm -rf ninja-1.9.0 ninja.zip
+RUN source /opt/rh/devtoolset-8/enable && \
+    curl -Ls https://github.com/ninja-build/ninja/archive/v1.9.0.zip -o ninja.zip && \
+    unzip ninja.zip && \
+    cd ninja-1.9.0 && \
+    ./configure.py --bootstrap && \
+    cp ninja /usr/bin && \
+    cd .. && \
+    rm -rf /tmp/*
 
 # install cmake
-RUN curl -L https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz -o /tmp/cmake.tar.gz &&\
-    echo "563a39e0a7c7368f81bfa1c3aff8b590a0617cdfe51177ddc808f66cc0866c76  /tmp/cmake.tar.gz" > /tmp/cmake-sha.txt &&\
-    sha256sum -c /tmp/cmake-sha.txt &&\
-    cd /tmp && tar xf cmake.tar.gz &&\
-    cp -r cmake-3.13.4-Linux-x86_64/* /usr/local/ &&\
-    rm -rf cmake.tar.gz cmake-3.13.4-Linux-x86_64 cmake-sha.txt
+RUN curl -Ls https://github.com/Kitware/CMake/releases/download/v3.13.4/cmake-3.13.4-Linux-x86_64.tar.gz -o cmake.tar.gz && \
+    echo "563a39e0a7c7368f81bfa1c3aff8b590a0617cdfe51177ddc808f66cc0866c76  cmake.tar.gz" > cmake-sha.txt && \
+    sha256sum -c cmake-sha.txt && \
+    mkdir cmake && \
+    tar --strip-components 1 --no-same-owner --directory cmake -xf cmake.tar.gz && \
+    cp -r cmake/* /usr/local/ && \
+    rm -rf /tmp/*
 
 # install LLVM
-RUN curl -L https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/llvm-project-11.0.0.tar.xz > /tmp/llvm.tar.xz
-RUN cd tmp &&\
-	echo "b7b639fc675fa1c86dd6d0bc32267be9eb34451748d2efd03f674b773000e92b  llvm.tar.xz" > llvm-sha.txt &&\
-	sha256sum -c llvm-sha.txt
-RUN cd /tmp && tar xf llvm.tar.xz --no-same-owner
-RUN mkdir /tmp/build && cd /tmp/build &&\
-	scl enable devtoolset-8 -- cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF \
-	-DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;compiler-rt;libcxx;libcxxabi;libunwind;lld;lldb"\
-	-DLLVM_STATIC_LINK_CXX_STDLIB=ON ../llvm-project-11.0.0/llvm
-RUN cd /tmp/build && cmake --build .
-RUN cd /tmp/build && cmake --build . --target install
-RUN rm -rf /tmp/*
+RUN source /opt/rh/devtoolset-8/enable && \
+    curl -Ls https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/llvm-project-11.0.0.tar.xz -o llvm.tar.xz && \
+    echo "b7b639fc675fa1c86dd6d0bc32267be9eb34451748d2efd03f674b773000e92b  llvm.tar.xz" > llvm-sha.txt && \
+    sha256sum -c llvm-sha.txt && \
+    mkdir llvm && \
+    tar --strip-components 1 --no-same-owner --directory llvm -xf llvm.tar.xz && \
+    mkdir build && \
+    cd build && \
+    cmake \
```

Comment:
I think doing `-G Ninja` will speed this up significantly (`cmake --build` will call into `make` otherwise and make is much slower than ninja)

Reply:
1000% agree, but I cant get ninja to build it, nor would make -j `nproc`. I'll gladly take suggestions input, because it is, in fact, awfully slow to build that step.

Reply:
this is super weird because this was definitely building in parallel on my machine. I will test this and see whether I can get this working (I imagine without parallelism this takes hours to build).

Reply:
@ammolitor  I changed the cmake call and just added `-G Ninja` and everything compiled just fine (the docker image is not built yet but it's past the llvm-step). Can you share the error you're seeing? It's super weird that this doesn't work for you (one difference is that I use podman because docker is stupid -- but in theory this shouldn't make a difference).

Reply:
Awesome, I'll try it again here too.

Reply:
I'm not going to pretend that I can construct an intelligent explanation for what I was doing wrong previously, but this works fine with '-G Ninja', I've updated as such.